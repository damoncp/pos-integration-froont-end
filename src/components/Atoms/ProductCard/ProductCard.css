.product-card {
  margin: .5rem;
}

.product-card .card-body {
  padding: 1rem .25rem .5rem;
}

.product-card__image {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.product-card__title {
  text-align: left;
  font-size: .8rem;
  font-weight: bold;
}

.product-card__image,
.product-card__title {
  width: 100%;
}
