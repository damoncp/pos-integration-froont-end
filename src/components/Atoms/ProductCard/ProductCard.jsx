import './ProductCard.css';
import React from 'react';
import { Card, CardImg, CardBody, CardTitle } from 'reactstrap';
import PropTypes from 'prop-types';

export default function ProductCard({ price, image, title }) {
  return (
    <Card className="product-card">
      <CardImg top
        className="product-card__image"
        data-testid="product-card-image"
        src={image} alt={title}
      />
      <CardBody>
        <CardTitle tag="h5" className="product-card__title">
          <span data-testid="product-card-price">${price}</span>
          {' - '}
          <span data-testid="product-card-title">{title}</span>
        </CardTitle>
      </CardBody>
    </Card>
  )
};

ProductCard.propTypes = {
  src: PropTypes.string,
  title: PropTypes.string,
  price: PropTypes.number,
};
