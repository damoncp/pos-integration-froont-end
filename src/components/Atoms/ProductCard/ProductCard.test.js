import { render, screen, cleanup } from '@testing-library/react';
import ProductCard from './ProductCard';

describe('Test <ProductCard /> component', () => {
  let renderProductCard;

  beforeEach(() => {
    renderProductCard = render(<ProductCard
      price={12}
      title={"Card title demo"} 
      image={"https://media.istockphoto.com/photos/bacon-burger-picture-id520215281?s=612x612"}
    />);
  });

  afterEach(cleanup);

  it('should render product card image', () => {
    // Arrange
    const productCardImageNode = screen.getByTestId('product-card-image');
    const productCardPriceNode = screen.getByTestId('product-card-price');
    const productCardTitleNode = screen.getByTestId('product-card-title');
    const productCardPrice = productCardPriceNode.innerHTML;
    const productCardTitle = productCardTitleNode.innerHTML;
    // Act

    // Assert
    expect(productCardImageNode).toHaveAttribute('src');
    expect(productCardImageNode.src).toBe('https://media.istockphoto.com/photos/bacon-burger-picture-id520215281?s=612x612');
    expect(productCardPrice).toEqual('$12');
    expect(productCardTitle).toEqual('Card title demo');
  });

  it('Products Snapshot test', () => {
    const { asFragment } = renderProductCard;

    expect(asFragment(renderProductCard)).toMatchSnapshot()
  });
});
