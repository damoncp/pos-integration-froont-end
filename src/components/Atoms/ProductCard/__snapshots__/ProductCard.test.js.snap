// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <ProductCard /> component Products Snapshot test 1`] = `
<DocumentFragment>
  <div
    class="product-card card"
  >
    <img
      alt="Card title demo"
      class="product-card__image card-img-top"
      data-testid="product-card-image"
      src="https://media.istockphoto.com/photos/bacon-burger-picture-id520215281?s=612x612"
    />
    <div
      class="card-body"
    >
      <h5
        class="product-card__title card-title"
      >
        <span
          data-testid="product-card-price"
        >
          $12
        </span>
         - 
        <span
          data-testid="product-card-title"
        >
          Card title demo
        </span>
      </h5>
    </div>
  </div>
</DocumentFragment>
`;
